### ISSUE TITLE
Consider more gentle directories

### ISSUE THREAD

**Issue by vanrein** [arpa2/mkhere#12]
The working directories `dl`, `src`, `build` are created in the build system root.  It may be neater to put them elsewhere, perhaps in `/var/cache/mkhere`.  This could be a variable in `lib/stdlib` that is set to `/` by default and that is not influenced by the environment.  Or, to avoid accidentally checking it in, it could be a (relative?) value stored (as a link?) in `lib/working_directory` and harvested via `readlink` with `/` as a fallback directory.

**Comment by vanrein** [comment]
We *might* even use this (or a similar) mechanism to move `/mkhere` into another directory.

**Comment by vanrein** [comment]
```
CACHEBASEDIR=$(readlink "$0/lib/cachebasedir" || echo '')
BUILDROOTDIR=$(readlink "$0/lib/buildrootdir" || dirname "$0")
```

**Comment by vanrein** [comment]
Avoid creating multiple directories too; initially there is no version.  Maybe defer directory creation to the main routine in `lib/stdlib`, when a `$VERSION` may have been set by the packaging script.

**Comment by vanrein** [comment]
The `readlink` sources can be included into `.gitignore` to stop them from being posted between users.

**Comment by vanrein** [comment]
Cannot get `.gitignore` to work... as usual?!?

**Comment by vanrein** [comment]
Resolved in fd41733a0e2daeb93ee088846f78255c92f5efff.
